### PR TITLE
Detail zonal provisioning for GCE PD

### PR DESCRIPTION
**Pull request by mattcary**
See https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/1005 for the context. This change clarifies how zonal provisioning will error depending on the state of CSI migration.

/cc @msau42 
/cc @jsafrane

### CODE REVIEW

File: content/en/docs/concepts/storage/storage-classes.md [hunk 1]
```diff
@@ -333,7 +333,11 @@ using `allowedTopologies`.
 
 {{< note >}}
 `zone` and `zones` parameters are deprecated and replaced with
-[allowedTopologies](#allowed-topologies)
+[allowedTopologies](#allowed-topologies). When [GCE CSI Migration](
+https://kubernetes.io/docs/concepts/storage/volumes/#gce-csi-migration) is
+enabled, a GCE PD volume can be provisioned in a topology that does not match
+any nodes. With legacy pre-migration GCE PD, in this case an error will be produced
```
Comment (msau42) [inline content/en/docs/concepts/storage/storage-classes.md:339]: maybe clarify that with csi migration, this will result in the pod failing to schedule.

Comment (mattcary) [reply content/en/docs/concepts/storage/storage-classes.md]: good idea, done.

File: content/en/docs/concepts/storage/storage-classes.md [hunk 2]
```diff
@@ -333,7 +333,13 @@ using `allowedTopologies`.
 
 {{< note >}}
 `zone` and `zones` parameters are deprecated and replaced with
-[allowedTopologies](#allowed-topologies)
+[allowedTopologies](#allowed-topologies). When [GCE CSI Migration](
+https://kubernetes.io/docs/concepts/storage/volumes/#gce-csi-migration) is
```
Comment (sftim) [inline content/en/docs/concepts/storage/storage-classes.md:337]: This should be:
```suggestion
[allowedTopologies](#allowed-topologies). When
[GCE CSI Migration](/docs/concepts/storage/volumes/#gce-csi-migration) is
```

File: content/en/docs/concepts/storage/storage-classes.md [hunk 3]
```diff
@@ -333,7 +333,13 @@ using `allowedTopologies`.
...
+enabled, a GCE PD volume can be provisioned in a topology that does not match
+any nodes, but any pod trying to use that volume will fail to schedule. With
+legacy pre-migration GCE PD, in this case an error will be produced
+instead at provisioning time. GCE CSI Migration is enabled by default beginning
+in 1.23.
```
Comment (sftim) [inline content/en/docs/concepts/storage/storage-classes.md:342]:
```suggestion
from the Kubernetes 1.23 release.
```

This avoids having a period that follows a version number.

**Review by sftim — COMMENTED** [review]
Thank you for the PR.

LGTM but with nits.

Aside: the reference for parameters that apply to different in-tree integations could belong within https://kubernetes.io/docs/reference/

(this PR is _not_ the right place to make that change)

**Comment by mattcary** [comment]
> 
> Aside: the reference for parameters that apply to different in-tree integations could belong within https://kubernetes.io/docs/reference/
> 
I agree, having this reference information here is a little weird. We'd want to change that for all the volume types though I think. Also migration is not very well documented.

**Comment by divya-mohan0209** [comment]
Hello folks, good day! Just checking on what would be the next steps for this PR?

**Comment by sftim** [comment]
I think this is ready for approval.
/lgtm

File: content/en/docs/concepts/storage/storage-classes.md [hunk 4]
```diff
@@ -333,7 +333,13 @@ using `allowedTopologies`.
 
 {{< note >}}
 `zone` and `zones` parameters are deprecated and replaced with
-[allowedTopologies](#allowed-topologies)
+[allowedTopologies](#allowed-topologies). When
+[GCE CSI Migration](/docs/concepts/storage/volumes/#gce-csi-migration) is
+enabled, a GCE PD volume can be provisioned in a topology that does not match
+any nodes, but any pod trying to use that volume will fail to schedule. With
+legacy pre-migration GCE PD, in this case an error will be produced
```
Comment (kbhawkey) [inline content/en/docs/concepts/storage/storage-classes.md:340]: Clarify: What is legacy pre-migration GCE PD?

When GCE CSI Migration is enabled, you can provision a GCE PD volume in a topology that does not match any nodes. Any pod trying to use the GCE PD volume fails to schedule. If you provision a legacy, pre-migration GCE PD volume, an error occurs.

**Pull request merged**